test/lib/general.lib.test.js

```diff
@@ -48,4 +48,49 @@ describe('RequestLib', () => {
       expect(result).to.equal('2015-10-21T20:31:57.000Z')
     })
   })
+
+  describe('#periodsOverlap', () => {
+    let referencePeriod
+    let checkPeriod
+
+    describe('when given dates that do not overlap', () => {
+      beforeEach(() => {
+        referencePeriod = [{
+          startDate: new Date('2023-02-01'),
+          endDate: new Date('2023-02-02')
+        }]
+
+        checkPeriod = [{
+          startDate: new Date('2023-01-01'),
+          endDate: new Date('2023-01-31')
+        }]
+      })
+
+      it('returns false', () => {
+        const result = GeneralLib.periodsOverlap(referencePeriod, checkPeriod)
+
+        expect(result).to.equal(false)
+      })
+    })
+
+    describe('when given dates that do overlap', () => {
```

Comment:
Because we name the params `fooPeriods` I'd suggest tweaking the `describe`

```suggestion
    describe('when given periods that do overlap', () => {
```